backend/common/src/main/java/ai/verta/modeldb/common/exceptions/ModelDBException.java

```diff
@@ -27,21 +29,25 @@ public ModelDBException(String message, Throwable cause) {
     code = Code.INTERNAL;
   }
 
+  @Deprecated(forRemoval = true)
   public ModelDBException(Throwable cause) {
     super(cause);
     code = Code.INTERNAL;
   }
 
+  @Deprecated(forRemoval = true)
   public ModelDBException(String message, Code code) {
     super(message);
     this.code = code;
   }
 
+  @Deprecated(forRemoval = true)
   public ModelDBException(String message, com.google.rpc.Code code, Throwable cause) {
     super(message, cause);
     this.code = Code.valueOf(code.name());
   }
 
+  @Deprecated(forRemoval = true)
   public ModelDBException(String message, com.google.rpc.Code code) {
     super(message);
     this.code = Code.valueOf(code.name());
```

Comment:
@conradoverta I think this will be needed because we are throwing some GRPC code like Code.ALREADY_EXISTS, Code.NOT_FOUND etc. and we are managing it on global exception. So if we deprecate it then what is the alternative of the same?

Reply:
I'm not sure I follow. We should be able to create the type-specific exceptions, and have a static method to convert a grpc error to the correct exception type.

Reply:
Ok then it makes sense, So you mean Instead of a common ModelDB exception, we have multiple exceptions for each type? right?

Reply:
We already have them, yes. :)